pkg/collector/corechecks/containers/kubelet/provider/cadvisor/provider_test.go

```diff
@@ -107,7 +108,8 @@ func (suite *ProviderTestSuite) SetupTest() {
 	fakeTagger := taggerimpl.SetupFakeTagger(suite.T())
 	defer fakeTagger.ResetTagger()
 	for entity, tags := range commontesting.CommonTags {
-		fakeTagger.SetTags(entity, "foo", tags, nil, nil, nil)
+		entityID, _ := taggertypes.NewEntityIDFromString(entity)
```

Comment:
Would it make sense to change the type and the definition of `commontesting.CommonTags` to change the type of keys from `string` to `EntityID` so that we don’t have to call `NewEntityIDFromString` here?

This is a test, so we don’t care about performances. But the goal of this PR is to reduce the usage of structured `EntityID` to and from `string` conversions and the other way round.
And I fear that any piece of code, including tests, can be used as a source of inspiration for developers writing pieces of code invoked in the hot path.
That’s why I think it would make sense to get rid of `NewEntityIDFromString` invocations in tests as well.

Reply:
My intent was to keep unit tests easy to write by supplying plain strings in each testcase. 

But I also agree it might serve as an inspiration for other developers to use `NewEntityIDFromString`, will update this. 

Thanks 👍